src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,208 @@
+
+            alterTask(input, Parser.TaskAction.DONE);
+
+        } else if (input.startsWith("delete")) {
+
+            // If successfully deleted task, then print current number of tasks.
+            if (alterTask(input, Parser.TaskAction.DELETE)) {
+                taskList.printNumberOfTasks();
+            }
+
+        } else {
+            vetoTask(input);
+        }
+
+        Ui.printDoubleDivider();
+        return true;
+    }
+
+    private void findTask(String input) {
+        try {
+            System.out.println("Output: These tasks have descriptions that contain the phrase '"
+                    + input.substring(5) + "'!\n");
+            taskList.printAllContains(input.substring(5));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("Output: Please include a search term after the word 'find'.\n"
+                    + "i.e. Find meeting");
+        }
+    }
+
+    /**
+     * Alters the TaskList depending on the TaskAction given, which could either be DELETE or DONE.
+     *
+     * @param input Input string from the user.
+     * @param action TaskAction to be done.
+     * @return True if the TaskList is successfully altered, false otherwise.
+     */
+    private boolean alterTask(String input, Parser.TaskAction action) {
+        try {
+
+            Task taskToBeAltered;
+
+            int index = Integer.parseInt(input.substring(action.substringIndex)) - 1;
+
+            if (action == Parser.TaskAction.DELETE) {
+                // If delete
+                taskToBeAltered = taskList.removeTask(index);
+            } else {
+                // If done
+                taskToBeAltered = taskList.markDoneInTaskList(index);
+            }
+
+            System.out.println("Output:\n\nThis task is successfully " + action.successMessage + ":\n\n"
+                    + "    " + taskToBeAltered);
+            return true;
+
+        } catch (StringIndexOutOfBoundsException | NumberFormatException e1) {
+
+            System.out.println("Output: Please specify which task you would like to have\n"
+                    + action.successMessage + " by adding a single number after '" + action.name + "'!\n"
+                    + "i.e. " + action.name + " 1");
+            return false;
+
+        } catch (IndexOutOfBoundsException e2) {
+
+            System.out.println("Output: There is no task under that number!");
+            return false;
+
+        }
+    }
+
+    /**
+     * Checks the task that is being attempted to be added into the task list.
+     * If the input contains the appropriate information, the task is added.
+     *
+     * @param input Input string from the user.
+     */
+    private void vetoTask(String input) {
+        Task newTask = null;
+
+        try {
+            if (!input.startsWith("todo") && !input.startsWith("deadline") && !input.startsWith("event")) {
+                throw new duke.exception.InvalidInputException();
+            } else {
```

Comment:
I think you can remove this `else`. If the `if` condition is satisfied, the program will directly go to the catch and skip all the code

Reply:
Done!